src/services/AuthService.tsx

```diff
@@ -223,46 +242,41 @@ const Auth = ({ children }: GenericContextProviderProps) => {
 					setRefreshInterval(getRefreshDelay(accessExp));
 					enqueueSnackbar('Logged in.', { variant: 'success' });
 				})
-				.catch((_: HTTPError) => {});
+				.catch((_: HTTPError) => {
+					enqueueSnackbar('Login failed.', { variant: 'error' });
+					setRefreshInterval(undefined);
+					logout();
+				});
 		},
-		[enqueueSnackbar, kyRef]
+		[enqueueSnackbar, kyRef, logout]
 	);
 
-	const logout = useCallback(() => {
-		if (user?.source === 'keycloak') keycloak.logout();
-		kyRef
-			.delete(`auth/logout`)
-			.json<YaptideResponse>()
-			.catch((_: HTTPError) => {})
-			.finally(() => setUser(null));
-	}, [kyRef, user?.source]);
-
 	const tokenRefresh = useCallback(() => {
-		if (!keycloak.authenticated) {
-			setKeyCloakInterval(undefined);
-
-			return Promise.resolve();
-		}
-
 		return keycloak
 			.updateToken(300) // 5 minutes in seconds minimum remaining lifetime for token before refresh is allowed
```

Comment:
Probably should be in external var

Reply:
This is based on configuration of the keycloak server token lifespan. We can't query for a value they use and this is the only place it matters. I see no reason to move it to a variable.